[Export, AOTInductor] Populate ShapeEnv's var_to_val during deserialization

SherlockNoMad opened the pull request:
Summary:
Deserialization didn't populate ShapeEnv's `var_to_val` field properly, and AOTInductor is relying on this field to compile dynamic shape properly. 
As a result, when AOTI failed at compiling a deserialized ExportedProgram.

Test Plan: buck2 test  mode/dev-nosan caffe2/test/inductor/fb:test_aot_inductor_pt2_inference

Differential Revision: D54559494

avikchaudhuri commented on torch/fx/experimental/symbolic_shapes.py:
```diff
@@ -2750,6 +2750,9 @@ def create_symbol(
             symbolic_context.shape_env_to_source_to_symbol_cache[id(self)][source_name] = r
         return r
 
+    def create_named_symbol_with_val(self, expr: sympy.Symbol, val: int):
```
This seems like a confusing name, since there are existing APIs like `create_symbol`. Maybe something explicit would be better, like `add_var_to_val`?

Also, should you not check whether the symbol already exists? (Probably doesn't matter but still...)

SherlockNoMad replied:
fixed.

SherlockNoMad committed:
[Export, AOTInductor] Populate ShapeEnv's var_to_val during deserialization (#121759)

Summary:

Deserialization didn't populate ShapeEnv's `var_to_val` field properly, and AOTInductor is relying on this field to compile dynamic shape properly. 
As a result, when AOTI failed at compiling a deserialized ExportedProgram.

Test Plan: buck2 test  mode/dev-nosan caffe2/test/inductor/fb:test_aot_inductor_pt2_inference

Reviewed By: avikchaudhuri

Differential Revision: D54559494

osalpekar commented:
@pytorchbot merge -f "unrelated failure"

osalpekar commented:
@pytorchbot merge -f "force merging this, fwd-fix incoming"

malfet commented:
@pytorchbot revert -m "Broke linux docs" -c ignoredsignal

malfet commented:
Cancelled, I see fwd fix
